cherro-front/src/pages/login/login.test.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-unused-vars */
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount, shallow } from 'enzyme';
+import { Login } from './login';
+import { LoginController } from '../../networking/controllers/login-controller';
+import { Router } from '../../routes';
+import { routeConfig } from '../../route-components';
+
+jest.mock('../../networking/controllers/login-controller');
+
+const faker = require('faker');
+
+describe('Login', () => {
+  const setupTest = async (nick, pass, output) => {
+    const nickField = output.find({ label: 'Nickname' }).first();
+    const passField = output.find({ label: 'Password' }).first();
+
+    await act(async () => {
+      nickField.props().onChange({ target: { value: nick } });
+      passField.props().onChange({ target: { value: pass } });
+    });
+
+    output.update();
+
+    return output;
+  };
+
+  it('should render correctly', () => {
+    const output = shallow(<Login />);
+    expect(output.exists()).toBe(true);
+  });
+
+  describe('when nickname and password are entered', () => {
+    it('login is enabled', async () => {
```

Comment:
Phrasing of test descriptors is important to allow easier reading of test suites. You can read more [here](https://www.bignerdranch.com/blog/why-do-javascript-test-frameworks-use-describe-and-beforeeach/)

```suggestion
    it('has an enabled submit button', async () => {
```

This comment also applies to other `it` calls 😄